[github-actions] meshcop test fails intermittently

[logs_7747.zip](https://github.com/openthread/ot-br-posix/files/6389182/logs_7747.zip)

```
otbr-agent[19504]: Set state callback: OK
otbr-agent[19504]: stop Thread Border Agent
otbr-agent[19504]: Initialize OpenThread Border Router Agent: OK
otbr-agent[19504]: dbus error (null): (null)
otbr-agent[19504]: InitializeListenFd error bind : Address already in use
otbr-agent[19504]: FAILED InitializeListenFd:158 - otbr rest server init error
+ pidof otbr-agent
+ die 'AGENT: failed to start'
+ exit_message='AGENT: failed to start'
+ echo ' *** ERROR: AGENT: failed to start'
 *** ERROR: AGENT: failed to start
+ exit 1
+ test_teardown
+ readonly EXIT_CODE=1
+ EXIT_CODE=1
+ write_syslog 'EXIT 1 - output logs'
+ logger -s -p syslog.alert 'OPENTHREAD_TEST: EXIT 1 - output logs'
<41>Apr 28 04:21:27 runner: OPENTHREAD_TEST: EXIT 1 - output logs
+ sudo pkill -f otbr-agent
+ sudo pkill -f otbr-web
+ sudo pkill -f commissioner-cli
+ sudo pkill -f ot-cli-ftd
+ wait
+ [[ 1 != 1 ]]
+ echo 'EXIT 1: MESSAGE: AGENT: failed to start'
EXIT 1: MESSAGE: AGENT: failed to start
+ exit 1
Error: Process completed with exit code 1.
```

--- a/src/rest/connection.cpp
+++ b/src/rest/connection.cpp
@@ -64,6 +64,11 @@ Connection::Connection(steady_clock::time_point aStartTime, Resource *aResource,
 {
 }
 
+Connection::~Connection(void)
+{
+    Disconnect();
+}
+
 void Connection::Init(void)
 {
     mParser.Init();

--- a/src/rest/connection.hpp
+++ b/src/rest/connection.hpp
@@ -65,6 +65,12 @@ public:
     Connection(steady_clock::time_point aStartTime, Resource *aResource, int aFd);
 
     /**
+     * The desctructor destroys the connection instance.
+     *
+     */
+    ~Connection(void) override;
+
+    /**
      * This method initializes the connection.
      *
      *

--- a/src/rest/rest_web_server.cpp
+++ b/src/rest/rest_web_server.cpp
@@ -52,6 +52,14 @@ RestWebServer::RestWebServer(ControllerOpenThread *aNcp)
 {
 }
 
+RestWebServer::~RestWebServer(void)
+{
+    if (mListenFd != -1)
+    {
+        close(mListenFd);
+    }
+}
+
 RestWebServer *RestWebServer::GetRestWebServer(ControllerOpenThread *aNcp)
 {
     static RestWebServer *sServer = new RestWebServer(aNcp);

--- a/src/rest/rest_web_server.hpp
+++ b/src/rest/rest_web_server.hpp
@@ -65,6 +65,12 @@ public:
     static RestWebServer *GetRestWebServer(ControllerOpenThread *aNcp);
 
     /**
+     * The destructor destroys the server instance.
+     *
+     */
+    ~RestWebServer(void) override;
+
+    /**
      * This method initializes the REST server.
      *
      */

--- a/tests/scripts/meshcop
+++ b/tests/scripts/meshcop
@@ -269,6 +269,7 @@ ba_start()
 
     write_syslog "AGENT: kill old"
     sudo killall "${OTBR_AGENT}" || true
+    sleep 5
     write_syslog "AGENT: starting"
 
     # we launch this in the background

